generators/client/files/state.js

```diff
@@ -0,0 +1,12 @@
+
+function load() {
+  try {
+    const element = document.getElementById('state');
+    return element ?
+      JSON.parse(element.textContent || element.innerText || '{}') : { };
+  } catch (e) {
+    return { };
+  }
+}
+
+export default load();
```

Comment:
I thought about this and my present rationale is that JSON is non-executable. This means that there are less concerns about escaping issues (it gets escaped anyway but preference for "safer" code) and I believe it means that CSR doesn't have to sign the block since it's non-executable. It also keeps the amount of generated data minimal.